src/groupeddataframe/grouping.jl

```diff
@@ -1106,3 +1108,16 @@ function _aggregate(d::AbstractDataFrame, fs::AbstractVector,
     sort && sort!(res, headers)
     res
 end
+
+function DataFrame(gd::GroupedDataFrame)
+    length(gd) == 0 && return similar(parent(gd), 0)
+    idx = similar(gd.idx)
+    doff = 1
+    for (s,e) in zip(gd.starts, gd.ends)
```

Comment:
Would a combination of `Iterators.flatten` and `copyto!` work? Not a big deal, but I wonder whether there's a simpler way of doing this relatively common operation.

Reply:
It could be made simpler (and it was simpler in DataFramesMeta.jl) but the approach I propose should be faster. Also I do not see how `Iterators.flatten` would be faster. In the approach I proposed we can use the fact that we have a small number of continuous blocks of memory.

Did you have in mind some alternative efficient pattern?

Reply:
I wouldn't expect it to be faster, just simpler. But I don't really care.